components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -102,6 +102,7 @@ class DefaultTopSitesStorage(
             try {
                 providerTopSites = topSitesProvider
                     .getTopSites(allowCache = true)
+                    .filter { providerConfig.providedTileFilter?.invoke(it) ?: true }
```

Comment:
I am not quite sure if this is a good idea yet, but I am thinking we should just change this to a generic `postProcess(it)` type of function so we aren't limited to just filtering, but can do any kind of post processing on the list of provided top sites. I am not 100% on the naming, but it does follow closely with the current middleware however there could be a better name for this as well.

Also, we would want to avoid naming this `tile` since that's Fenix specific. We should instead just say `providerTopSitesFilter`.

Reply:
I changed the function name to `topSitesProviderFilter`, following the config naming (`TopSitesProviderConfig`). 

I think we should stick with the filtering approach. I don't see any other kind of post processing that would be required inside `DefaultTopSitesStorage`, since the `postProcess` function would be called on a `List<TopSites>`.